content/guides/fine-grained-reactivity.mdx

```diff
@@ -0,0 +1,288 @@
+  
+  function getter() {
+    return value;
+  }
+  
+  function setter(newValue) {
+    value = newValue;
+  }
+  
+  return [getter, setter];
+}
+
+// ..
+```
+
+This allows you to retrieve the current value through the getter and make any changes via the setter.
+At this stage, reactivity is not present, however. 
+
+### `createEffect`
+
+`createEffect` defines a function that immediately calls the function that is passed into it:
+
+```jsx
+// ..
+
+function createEffect(fn) {
+  fn();
+}
+
+// ..
+```
+
+### Making a system reactive
+
+Reactivity emerges when linking `createSignal` and `createEffect` and this happens through:
+
+1. Maintaining a reference to the current subscriber's function.
+2. Registering this function during the creation of an effect.
+3. Adding the function to a subscriber list when accessing a signal.
+4. Notifying all subscribers when the signal has updated.
+
+```jsx
+let currentSubscriber = null; 
+
+function createSignal(initialValue) {
+  let value = initialValue;
+  const subscribers = new Set();
+  
+  function getter() {
+    if (currentSubscriber) {
+      subscribers.add(currentSubscriber);
+    }
+    return value;
+  }
+  
+  function setter(newValue) {
+    if (value === newValue) return; // if the new value is not different, do not notify dependent effects and memos
+    value = newValue;
+    for (const subscriber of subscribers) {
+      subscriber(); // 
+    }
+  }
+  
+  return [getter, setter];
+}
+
+// creating an effect
+function createEffect(fn) {
+  const previousSubscriber = currentSubscriber; // Step 1
+  currentSubscriber = fn;
+  fn();
+  currentSubscriber = previousSubscriber;
+}
+
+//.. 
+```
+
+A variable is used to hold a reference to the current executing subscriber function.
+This is used to determine which effects are dependent on which signals.
+
+Inside `createSignal`, the initial value is stored and a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) is used to store any subscriber functions that are dependent on the signal.
+This function will then return two functions for the signal:
+* The `getter` function checks to see if the current subscriber function is being accessed and, if it is, adds it to the list of subscribers before returning the *current* value of the signal.
+* The `setter` function evaluated the new value against the old value, notifying the dependent functions only when the signal has been updated.
+
+When creating the `createEffect` function, a reference to any previous subscribers is intiailized to handle any possible nested effects present.
+The current subscriber is then passed to the given function, which is run immediately.
+During this run, if the effect accesses any signals it is then registered as a subscriber to those signals.
+The current subscriber, once the given function has been run, will be reset to its previous value so that, if there are any nested effects, they are operated correctly.
```

Comment:
```suggestion
The current subscriber, once the given function runs, will be reset to its previous value so that, if there are any nested effects, they are operated correctly.
```